cloudinit/sources/DataSourceAzure.py

```diff
@@ -706,6 +722,9 @@ def crawl_metadata(self):
                 self._wait_for_pps_unknown_reuse()
 
             md, userdata_raw, cfg, files = self._reprovision()
+            if cfg.get("ProvisionGuestProxyAgent"):
+                self._check_azure_proxy_agent_status()
```

Comment:
No unittests cover this new PPS invocation. If possible can we extend tests/unittests/sources/test_azure::...test_*_pps tests with `construct_ovf_env(provision_guest_proxy_agent=True)` to ensure we get into _check_azure_proxy_agent_status.